Source/JavaScriptCore/assembler/MacroAssemblerX86_64.h

```diff
@@ -2140,9 +2140,8 @@ class MacroAssemblerX86_64 : public MacroAssemblerX86Common {
         m_assembler.pinsr(dest, src, simdLane, lane.m_value);
     }
 
-    void vectorReplaceLane(SIMDLane simdLane, TrustedImm32 lane, FPRegisterID src, FPRegisterID dest)
+    void vectorReplaceLane(SIMDLane simdLane, TrustedImm32 lane, FPRegisterID src, FPRegisterID dest, RegisterID scratch)
     {
-        RegisterID scratch = scratchRegister();
         moveDoubleTo64(src, scratch);
         m_assembler.pinsr(dest, scratch, simdLane, lane.m_value);
```

Comment:
Do we really need a scratch here? Intel has `(v)insertps` for inserting a scalar float into a vector: https://www.felixcloutier.com/x86/insertps

Reply:
Seems like it might be a bit more complicated than I'm thinking, we can evaluate this later.

Reply:
Added a FIXME.